projects/plugins/jetpack/_inc/lib/class-jetpack-partner-coupon-helper.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Class for the Jetpack partner coupon logic.
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Disable direct access.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Jetpack_Partner_Coupon_Helper
+ *
+ * @since 10.4.0
+ */
+class Jetpack_Partner_Coupon_Helper {
+
+	/**
+	 * Name of the Jetpack_Option coupon option.
+	 *
+	 * @var string
+	 */
+	private static $coupon_option = 'partner_coupon';
+
+	/**
+	 * Name of the Jetpack_Option added option.
+	 *
+	 * @var string
+	 */
+	private static $added_option = 'partner_coupon_added';
+
+	/**
+	 * Jetpack_Partner_Coupon_Helper
+	 *
+	 * @var Jetpack_Partner_Coupon_Helper|null
+	 **/
+	private static $instance = null;
+
+	/**
+	 * A list of supported partners.
+	 *
+	 * @var array
+	 */
+	private static $supported_partners = array(
+		'IONOS' => 'IONOS',
+	);
+
+	/**
+	 * A list of supported presets.
+	 *
+	 * @var array
+	 */
+	private static $supported_presets = array(
+		'IONA' => 'jetpack_backup_daily',
+	);
+
+	/**
+	 * Initialize class.
+	 */
+	public static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_Partner_Coupon_Helper();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	private function __construct() {
+		add_action( 'admin_init', array( $this, 'catch_coupon' ) );
+		add_action( 'admin_init', array( $this, 'purge_coupon' ) );
+	}
+
+	/**
+	 * Catch partner coupon and redirect to claim component.
+	 */
+	public function catch_coupon() {
+		// Accept and store a partner coupon if present, and redirect to Jetpack connection screen.
+		$partner_coupon = isset( $_GET['jetpack-partner-coupon'] ) ? sanitize_text_field( $_GET['jetpack-partner-coupon'] ) : false; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( $partner_coupon ) {
+			Jetpack_Options::update_options(
+				array(
+					self::$coupon_option => $partner_coupon,
+					self::$added_option  => gmdate( 'Y-m-d H:i:s' ),
+				)
+			);
+
+			if ( Jetpack::connection()->is_connected() ) {
```

Comment:
Yeah, it - especially - made sense when the class got moved to the Partner package because the `Jetpack` class only exists in the All-in-One Jetpack plugin